src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -25,10 +32,28 @@ public class CertificateRegistry {
             CertificateV1DTO.Status.ACTIVE, Certificate.Status.ACTIVE,
             CertificateV1DTO.Status.UNKNOWN, Certificate.Status.UNKNOWN
     );
+    private final ClientCertificateStore pemStore;
 
+    /**
+     * Creates a certificate registry.
+     * @param runtimeConfiguration - Runtime configuration
+     * @param kernel - Nucleus Kernel
+     *
+     * @throws IOException - If fails to get the service work path
+     * @throws KeyStoreException - If fails to create a key store to store certificate PEMs.
+     */
     @Inject
-    public CertificateRegistry(RuntimeConfiguration runtimeConfiguration) {
+    public CertificateRegistry(RuntimeConfiguration runtimeConfiguration, Kernel kernel) throws IOException,
+            KeyStoreException {
+       this(
+           runtimeConfiguration,
+           kernel.getNucleusPaths().workPath(ClientDevicesAuthService.CLIENT_DEVICES_AUTH_SERVICE_NAME)
+       );
+    }
+
+    public CertificateRegistry(RuntimeConfiguration runtimeConfiguration, Path pemStorePath) throws KeyStoreException {
```

Comment:
Why not inject the cert store?

Reply:
I guess I don't know? 😅. I think I was originally doing something else and then forgot